src/actions/performApiDataRequest.test.ts

```diff
@@ -59,70 +59,54 @@ test('It gets an error message when config is empty', () => {
     return expect(performApiRequest('postData/', {}, { data: 'json' })(dispatch, () => store.getState())).rejects.toBe('apiData.performApiRequest: no config with key postData/ found!');
 });
 
-test('The function resolves', () => {
-    state = { apiData: getState('postData', true, {}, 'ready', 'POST') };
-    return expect(performApiRequest('postData', {}, { data: 'json' })(dispatch, () => store.getState())).resolves.toBeUndefined();
-});
-
-test('The function resolves with custom response', () => {
-    state = { apiData: getState('postData', true, {}, 'ready', 'POST') };
-
-    return expect(performApiRequest('postData', {}, { data: 'json' })(dispatch, () => store.getState())).resolves.toBeUndefined();
-});
-
 (request as jest.Mock).mockImplementationOnce(() =>
     Promise.resolve(response1)
 );
-test('it calls ApiDataSuccess', async (done) => {
+
+test('The function resolves with custom response and calls apiDataSuccess', async (done) => {
     state = { apiData: getState('postData', true, {}, 'ready', 'POST') };
-    // @ts-ignore
-    expect(dispatch).toHaveBeenCalledWith(apiDataSuccess(getRequestKey('postData'), state.apiData.endpointConfig, response1.response, undefined));
+    expect(performApiRequest('postData', {}, { data: 'json' })(dispatch, () => store.getState())).resolves.toBeUndefined();
```

Comment:
Incorrect async test implementation. There should be an await before performApiRequest and the expect should be returned after that so jest knows the test is done, or done() can be called at the end with multiple expects.
Also, resolve value does not need to be checked in multiple tests.